src/components/PostDetails.tsx

```diff
@@ -1,118 +1,66 @@
-    setVisible(false);
-
-    commentsApi
-      .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
-    setVisible(false);
-    setError(false);
-
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
-    }
-  };
+  const { selectedPost: post } = useAppSelector(state => state.posts);
+  const { comments, status } = useAppSelector(state => state.comments);
+
+  const isLoaded = status !== LoadingStatus.Loading;
+  const hasError = status === LoadingStatus.Failed;
+
+  const [isVisible, setIsVisible] = useState(false);
 
   useEffect(() => {
-    loadComments();
-  }, []);
-
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
-
-  const addComment = async ({ name, email, body }: CommentData) => {
-    try {
-      const newComment = await commentsApi.createComment({
-        name,
-        email,
-        body,
-        postId: post.id,
-      });
-
-      setComments(currentComments => [...currentComments, newComment]);
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
-    } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
-    }
-  };
+    setIsVisible(false);
 
-  const deleteComment = async (commentId: number) => {
-    // we delete the comment immediately so as
-    // not to make the user wait long for the actual deletion
-    // eslint-disable-next-line max-len
-    setComments(currentComments =>
-      currentComments.filter(comment => comment.id !== commentId),
-    );
+    if (post) {
+      dispatch(getCommentsAsync(post));
+    } else {
+      dispatch(clearComments());
+    }
+  }, [dispatch, post]);
 
-    await commentsApi.deleteComment(commentId);
+  const deleteComment = async (comment: Comment) => {
+    dispatch(deleteCommentAsync(comment));
   };
 
   return (
     <div className="content" data-cy="PostDetails">
       <div className="block">
-        <h2 data-cy="PostTitle">{`#${post.id}: ${post.title}`}</h2>
+        <h2 data-cy="PostTitle">{`#${post?.id}: ${post?.title}`}</h2>
 
-        <p data-cy="PostBody">{post.body}</p>
+        <p data-cy="PostBody">{post?.body}</p>
       </div>
 
       <div className="block">
-        {!loaded && <Loader />}
+        {!isLoaded && <Loader />}
 
-        {loaded && hasError && (
+        {isLoaded && hasError && (
```

Comment:
The same issue. Change for all the cases
```suggestion
        {hasError && (
```